### PR TITLE
Add filtering to the cli based on api

### PR DESCRIPTION
**Pull request by jakub-musik**
If merged, this PR will

- add --filter option to the CLI and explain it in the docs

- refactor transformers to pass parameters as objects, accommodating future feature growth 

- implement a masking transformer and use it in both CLI and API

### CODE REVIEW

File: src/shared/translations/masked-translations.ts [hunk 1]
```diff
@@ -25,11 +23,7 @@ export default class MaskedTranslations implements ITranslations {
...
     }
 
-    const { tags, ...translations } = translationsWithTags;
-
-    const maskInput = this.maskInput.convert(filters);
-    const filtersMask = this.maskConverter.convert(maskInput, tags);
-    const maskedTranslations = mask(translations, filtersMask);
+    const maskedTranslations = this.jsonToJsonMaskedTransformer.transform(translationsWithTags, { filters });
 
     return Promise.resolve(maskedTranslations);
```
Comment (blyszcz) [inline src/shared/translations/masked-translations.ts]: async function will return Promise by default, no need to write `Promise.resolve()`

File: src/shared/transformers/spreadsheet-to-ios-strings.transformer.ts [hunk 2]
```diff
@@ -3,18 +3,30 @@ import ITransformer from './transformer';
...
+      filters,
+    }: {
+      langCode?: string;
+      mergeLanguages?: boolean;
+      filters?: string[];
+    } = {}
   ): string | object[] {
-    const json = this.spreadsheetToJson.transform(source, langCode);
+    let json = this.spreadsheetToJson.transform(source, { langCode });
+    json = this.jsonToJsonMasked.transform(json, { filters });
```
Comment (blyszcz) [inline src/shared/transformers/spreadsheet-to-ios-strings.transformer.ts]: `const json = ...`
`const maskedJson = ...`

File: src/shared/transformers/json-to-json-masked.transformer.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      mergeLanguages,
+      filters,
+    }: {
+      langCode?: string;
+      mergeLanguages?: boolean;
+      filters?: string[];
+    }
+  ): { [key: string]: object } {
+    const { tags, ...translations } = source;
+    if (filters && filters.length > 0) {
```
Comment (blyszcz) [inline src/shared/transformers/json-to-json-masked.transformer.ts]: `filters.length > 0` -> `filters.length`

File: src/shared/transformers/spreadsheet-to-ios-strings.transformer.ts (hunk 2 above)
Comment (barograf) [reply src/shared/transformers/spreadsheet-to-ios-strings.transformer.ts]: There are more places like that, please take a look at them too.

File: src/shared/translations/masked-translations.ts (hunk 1 above)
Comment (jakub-musik) [reply src/shared/translations/masked-translations.ts]: removed

File: src/shared/transformers/spreadsheet-to-ios-strings.transformer.ts (hunk 2 above)
Comment (jakub-musik) [reply src/shared/transformers/spreadsheet-to-ios-strings.transformer.ts]: replaced everywhere with consts

File: src/shared/transformers/json-to-json-masked.transformer.ts (hunk 3 above)
Comment (jakub-musik) [reply src/shared/transformers/json-to-json-masked.transformer.ts]: changed

**Pull request merged**